gradle/init.gradle.kts

```diff
@@ -1,32 +1,46 @@
-fun resolvePluginPath(rootPath: java.nio.file.Path): java.nio.file.Path? {
-    if (rootPath.resolve("src").toFile().exists()) {
-        return null
-    }
-    if (rootPath.resolve("demo").toFile().exists()) {
-        return null
-    }
-    var path = rootPath
-    for (n in 1..4) {
-        val settingsFile = path.resolve("demo")
-        if (settingsFile.toFile().exists()) {
-            return path.toAbsolutePath()
-            break
-        } else {
-            path = path.resolve("..")
-            println("new path: ${path.toAbsolutePath()}")
-        }
-    }
-    return null
-}
-
 gradle.settingsEvaluated {
-    val property = "org.jsonschema2dataclass.local"
-    if (extra.has(property) && extra[property].toString().toBoolean()) {
-        println("root dir: ${getRootDir()}")
-        val path = resolvePluginPath(getRootDir().toPath().toAbsolutePath())
-        if (path != null) {
-            println("Include build:\"$path\"")
-            includeBuild(path)
+    val localPublishExtra = "org.jsonschema2dataclass.internal.local-publish"
+    val localVersionExtra = "org.jsonschema2dataclass.internal.git-version"
+    if (extra.has(localPublishExtra) && extra[localPublishExtra].toString().toBoolean()) {
+        val localVersion = extra[localVersionExtra]!!.toString()
+        pluginManagement {
+            repositories {
+                exclusiveContent {
+                    mavenLocal()
+                    filter {
+                        includeGroupByRegex(".*.jsonschema2dataclass.*")
+                    }
+                }
+                mavenCentral()
+                gradlePluginPortal()
+            }
+            resolutionStrategy {
+                eachPlugin {
+                    if (requested.id.name == "jsonschema2dataclass" && requested.version != localVersion) {
+                        useModule("org.jsonschema2dataclass:plugin:$localVersion")
+                    }
+                }
+            }
+        }
+        rootProject {
+            buildscript {
+                repositories {
+                    exclusiveContent {
+                        mavenLocal()
+                        filter {
+                            includeGroupByRegex(".*.jsonschema2dataclass.*")
+                        }
+                    }
+                    mavenCentral()
+                }
```

Comment:
I think you need `forRepository` otherwise this is just the same as:
```
               repositories {
                    mavenLocal()
                    exclusiveContent {
                        filter {
                            includeGroupByRegex(".*.jsonschema2dataclass.*")
                        }
                    }
                    mavenCentral()
```
but I might be wrong

Reply:
yep, I've miss that